_posts/2020-04-20-schedule-your-functions.md

```diff
@@ -0,0 +1,130 @@
+
+## Prepare environment
+
+Before showing you how to run our functions on a schedule, first we need to set up the environment with the tools mentioned above.
+
+### Create cluster
+
+First thing we need is running Kubernetes cluster:
+
+```bash
+kind create cluster
+```
+
+Wait for the installation to finish and verify the pods in the `kube-system` namespace are `1/1`:
+
+```bash
+kubectl get pods -n kube-system
+```
+
+### Install OpenFaaS
+
+With `arkade` the installation of OpenFaaS boils down to single line command:
+
+```bash
+arkade install openfaas
+```
+
+Wait for the OpenFaaS gateway to be ready:
+
+```bash
+kubectl rollout status -n openfaas deploy/gateway
+```
+
+### Set up OpenFaaS client
+
+Get the OpenFaaS Gateway password:
+
+```bash
+PASSWORD=$(kubectl get secret -n openfaas basic-auth -o jsonpath="{.data.basic-auth-password}" | base64 --decode; echo)
+```
+
+Set up the gateway endpoint for the client:
+
+```bash
+export OPENFAAS_URL=http://$(docker inspect -f "{{ .NetworkSettings.IPAddress }}" $(docker ps -aqf "name=kind-control-plane")):31112
+```
+
+Authenticate the client:
+
+```bash
+echo -n $PASSWORD | faas-cli login --username admin --password-stdin
+```
+
+## Schedule the functions
+
+After we have the environment up and running, now we can install the cron-connector and schedule our functions to run at specific intervals.
+
+### Deploy the cron-connector
+
+In order to install our `cron-connector` and schedule a function run the following command:
+
+```bash
+curl -s https://raw.githubusercontent.com/openfaas-incubator/cron-connector/master/yaml/kubernetes/connector-dep.yml | kubectl apply -f -
+```
+
+Wait for the `cron-connector` to be deployed:
+
+```bash
+kubectl rollout status -n openfaas deploy/sample-connector
+```
+
+## Run the function
+
+We will run a function from the store once a minute with a `schedule` annotation following this cron syntax `*/1 * * * *` which is the same as in the [linux cron configuration](https://www.cyberciti.biz/faq/how-do-i-add-jobs-to-cron-under-linux-or-unix-oses/).
+
+The cron-connector recognizes which functions should be ran on a schedule by annotation `topic` with value `cron-function`.
+
+Finally we will deploy `node-info` function and apply the schedule and topic annotations mentioned above:
+
+```bash
+faas-cli store deploy nodeinfo --annotation schedule="*/1 * * * *" --annotation topic=cron-function
+```
+
+If you wait for a few minutes and visit the OpenFaaS UI you will see how the `Invocation count` of the function raises every minute.
+
+## Wrapping up
+
+In this post we have shown how you can run minimal OpenFaaS environment with simple function and how with the help of the cron-connector you can execute the function on a desired schedule.
+
+Now you can go ahead and [create your own function](https://docs.openfaas.com/cli/templates/) in the limitless amount of languages and make sure it always runs on time!
```

Comment:
"in the limitless amount of languages and make sure it always runs on time!" -> Can you let the reader decide if it's limitless?

How about:

```
Learn how to create your own functions with the [OpenFaaS workshop]. You'll find many templates in the template store, run `faas-cli template store list` or view the [Template Documentation]
```

Reply:
Changed as suggested :+1: